lib/wallet/http.js

```diff
@@ -1551,6 +1553,103 @@ class HTTP extends Server {
 
       return res.json(200, mtx.getJSON(this.network));
     });
+
+    // SendMany
+    this.post('/wallet/:id/sendmany', async (req, res) => {
+      const valid = Validator.fromRequest(req);
+      const account = valid.str('account', 'default');
+      const sendTo = valid.array('sendto', []);
+      const minconf = valid.u32('minconf', 1);
+      const subtractFee = valid.bool('subtractfee', false);
+      const passphrase = valid.str('passphrase');
+
+      if (sendTo.length === 0)
+        throw new Error('parameter sendto is required and can not be empty!');
+
+      const cache = req.wallet.hsdTransferResults;
+      const results = [];
+
+      for (const {idempotency_key: idempotencyKey} of sendTo) {
+        if (idempotencyKey && cache.has(idempotencyKey))
+          results.push(cache.get(idempotencyKey));
+      }
+
+      const filteredSendTo = sendTo.filter(element =>
+        !cache.has(element.idempotency_key));
+
+      const addressToIdempotencyKeyMap = new Map();
+      const uniq = new BufferSet();
+
+      const outputs = filteredSendTo.map((element) => {
+        const to = new Validator(element);
+
+        const idempotencyKey = to.str('idempotency_key');
+        if (!idempotencyKey)
+          throw new Error('idempotencyKey is missing!');
+
+        const value   = to.u64('value');
+        if (!value)
+          throw new Error(`value: ${value} is invalid!`);
+
+        const rawAddress = to.str('address');
+        const addr = parseAddress(rawAddress, this.network);
+        addressToIdempotencyKeyMap.set(addr.toString(), {
+          idempotencyKey: idempotencyKey,
+          rawAddress: rawAddress
+        });
+
+        const hash = addr.getHash();
+
+        if (uniq.has(hash))
+          throw new Error('Invalid parameter!');
```

Comment:
Perhaps indicate that it's a duplicate hash in the error message (and include the address that caused it).